### PR TITLE
Adding detailed parse validation

### LINKED ISSUE #591: PR #581 introduces a little bit of lag / overwriting when typing aggregration expression in metrics designer
Hey @AdityaHegde, while working on #590, I've noticed that there's just a little tiny bit of lag or some race condition that causes typing inputs to be dropped when I'm typing fast. It's not terrible for me, but it might be good to see if there is a simple solution. I suspect it is some race condition-ish thing introduced by the on:keyup added in PR https://github.com/rilldata/rill-developer/pull/581. Hopefully an easy fix! :-)

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #591 #589 #585
Adding human readable errors for measure expression parse failures.
Adding argument validation to some extent. Will fallback to validating by running the expression through duckdb.

### CODE REVIEW

File: src/common/expression-parser/validateAggregate.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  // number, any
+  arg_max: [NUMERICS, ANY_TYPES],
+  argMax: [NUMERICS, ANY_TYPES],
+  max_by: [NUMERICS, ANY_TYPES],
+  arg_min: [NUMERICS, ANY_TYPES],
+  argMin: [NUMERICS, ANY_TYPES],
+  min_by: [NUMERICS, ANY_TYPES],
+
+  // number
+  count: [NUMERICS],
```
Comment (bcolloran) [inline src/common/expression-parser/validateAggregate.ts:18]: i think we should be able to `count` and `count distinct` any type, right?

Comment (bcolloran) [reply src/common/expression-parser/validateAggregate.ts:18]: approx_count_distinct too?

**Review by bcolloran — APPROVED** [review]
It looks good @AdityaHegde  I'm approving, but I had one small question about the valid types in count expressions (I think anything should be allowed, right?), so I thought I'd let you double check that and you can merge tomorrow

File: src/common/expression-parser/validateAggregate.ts (hunk 1 above)
Comment (AdityaHegde) [reply src/common/expression-parser/validateAggregate.ts]: Good point. Fixed this used case.

**Pull request merged**